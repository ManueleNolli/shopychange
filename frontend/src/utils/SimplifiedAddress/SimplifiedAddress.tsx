import React from 'react'

export const simplifiedAddress = (address: string) => {
  return address.slice(0, 6) + '...' + address.slice(-4)
}

import { Link as LinkRouter } from 'react-router-dom'
import { Text, Link } from '@chakra-ui/react'

export const renderAddress = (address: string, navigate: string) => {
  return (
    <Link color="blue.500" as={LinkRouter} to={navigate}>
      <Text>{simplifiedAddress(address)}</Text>
    </Link>
  )
}

export const renderAddressWithTokenId = (
  address: string,
  tokenId: number,
  navigate: string
) => {
  return (
    <Link color="blue.500" as={LinkRouter} to={navigate}>
      <Text>
        {simplifiedAddress(address)}/{tokenId}
      </Text>
    </Link>
  )
}
