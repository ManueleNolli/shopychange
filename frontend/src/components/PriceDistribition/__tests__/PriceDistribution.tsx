import React from 'react'
import { act, render } from '@testing-library/react'
import PriceDistrubution from '../PriceDistribution'
import usePriceDistrubution from '../usePriceDistribution'
import { renderAddress } from '../../../utils/SimplifiedAddress/SimplifiedAddress'

jest.mock('../usePriceDistribution', () => jest.fn())

jest.mock('../../../utils/SimplifiedAddress/SimplifiedAddress', () => ({
  renderAddress: jest.fn(),
}))

describe('PriceDistrubution', () => {
  test('render correctly owner', async () => {
    ;(usePriceDistrubution as jest.Mock).mockReturnValue({
      settedPrice: 2,
      remainingValue: 1,
      otherRoyalties: [
        {
          account: 'mockedAccount',
          share: 10,
        },
      ],
      otherRoyaltiesSum: 0,
      marketplaceRoyalty: 5,
      fixedNumber: jest.fn(),
      isFetching: false,
      isError: false,
    })
    ;(renderAddress as jest.Mock).mockReturnValue('mockedAddress')
    const tree = render(
      <PriceDistrubution
        price={1}
        isOwner={true}
        address={'mockedAddress'}
        tokenId={0}
      />
    )
    await act(async () => {
      Promise.resolve()
    })

    expect(tree.container).toMatchSnapshot()
  })
  test('render correctly not owner', async () => {
    ;(usePriceDistrubution as jest.Mock).mockReturnValue({
      settedPrice: 2,
      remainingValue: 1,
      otherRoyalties: [],
      otherRoyaltiesSum: 0,
      marketplaceRoyalty: 5,
      fixedNumber: jest.fn(),
      isFetching: false,
      isError: false,
    })
    ;(renderAddress as jest.Mock).mockReturnValue('mockedAddress')

    const tree = render(
      <PriceDistrubution
        price={1}
        isOwner={false}
        address={'mockedAddress'}
        tokenId={0}
      />
    )
    await act(async () => {
      Promise.resolve()
    })

    expect(tree.container).toMatchSnapshot()
  })
})
