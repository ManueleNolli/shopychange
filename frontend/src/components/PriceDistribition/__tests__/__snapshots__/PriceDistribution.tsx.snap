// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceDistrubution render correctly not owner 1`] = `
<div>
  <div
    class="css-1txxpn"
  >
    <h2
      class="chakra-heading css-1fa2p9l"
    >
      Price Distribution
    </h2>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
    <div
      class="chakra-table__container css-178nss3"
    >
      <table
        class="chakra-table css-0"
      >
        <thead
          class="css-0"
        >
          <tr
            class="css-0"
          >
            <th
              class="css-0"
            >
              To
            </th>
            <th
              class="css-0"
            >
              Percentage
            </th>
            <th
              class="css-0"
            >
              Amount (eth)
            </th>
          </tr>
        </thead>
        <tbody
          class="css-0"
        >
          <tr
            class="css-0"
          >
            <td
              class="css-0"
            >
              Shopychange
            </td>
            <td
              class="css-0"
            >
              5
            </td>
            <td
              class="css-0"
            />
          </tr>
          <tr
            class="css-0"
          >
            <td
              class="css-0"
            >
              <i
                class="chakra-text css-1b1uf4q"
              >
                The remaining 
                95
                % is separated as follows
              </i>
            </td>
          </tr>
          <tr
            class="css-0"
          >
            <td
              class="css-0"
            >
              Owner
            </td>
            <td
              class="css-0"
            >
              100
            </td>
            <td
              class="css-0"
            />
          </tr>
        </tbody>
      </table>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
  </div>
</div>
`;

exports[`PriceDistrubution render correctly owner 1`] = `
<div>
  <div
    class="css-1txxpn"
  >
    <h2
      class="chakra-heading css-1fa2p9l"
    >
      Price Distribution
    </h2>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
    <div
      class="chakra-table__container css-178nss3"
    >
      <table
        class="chakra-table css-0"
      >
        <thead
          class="css-0"
        >
          <tr
            class="css-0"
          >
            <th
              class="css-0"
            >
              To
            </th>
            <th
              class="css-0"
            >
              Percentage
            </th>
            <th
              class="css-0"
            >
              Amount (eth)
            </th>
          </tr>
        </thead>
        <tbody
          class="css-0"
        >
          <tr
            class="css-0"
          >
            <td
              class="css-0"
            >
              Shopychange
            </td>
            <td
              class="css-0"
            >
              5
            </td>
            <td
              class="css-0"
            />
          </tr>
          <tr
            class="css-0"
          >
            <td
              class="css-0"
            >
              <i
                class="chakra-text css-1b1uf4q"
              >
                The remaining 
                95
                % is separated as follows
              </i>
            </td>
          </tr>
          <tr
            class="css-0"
          >
            <td
              class="css-0"
            >
              You
            </td>
            <td
              class="css-0"
            >
              100
            </td>
            <td
              class="css-0"
            />
          </tr>
          <tr
            class="css-0"
          >
            <td
              class="css-0"
            >
               
              mockedAddress
            </td>
            <td
              class="css-0"
            >
              10
            </td>
            <td
              class="css-0"
            />
          </tr>
        </tbody>
      </table>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
  </div>
</div>
`;
